roseau/load_flow/wrapper.py

```diff
@@ -0,0 +1,152 @@
+import collections
+import functools
+from collections.abc import Iterable
+from inspect import Parameter, Signature, signature
+from itertools import zip_longest
+from typing import Any, Callable, Optional, TypeVar, Union
+
+from pint import Quantity, Unit
+from pint.registry import UnitRegistry
+from pint.util import to_units_container
+
+T = TypeVar("T")
+FuncT = TypeVar("FuncT", bound=Callable)
+
+
+def _parse_wrap_args(args: Iterable[Optional[Union[str, Unit]]]) -> Callable:
+    """Create a converter function for the wrapper"""
+    # Arguments which have units.
+    unit_args_ndx = set()
+
+    # _to_units_container
+    args_as_uc = [to_units_container(arg) for arg in args]
+
+    # Check for references in args, remove None values
+    for ndx_, arg in enumerate(args_as_uc):
+        if arg is not None:
+            unit_args_ndx.add(ndx_)
+
+    def _converter(ureg: UnitRegistry, sig: Signature, values: list[Any], kw: dict[Any]):
+        len_initial_values = len(values)
+
+        # pack kwargs
+        for i, param_name in enumerate(sig.parameters):
+            if i >= len_initial_values:
+                values.append(kw[param_name])
+
+        # convert arguments
+        for ndx in unit_args_ndx:
+            if isinstance(values[ndx], ureg.Quantity):
+                values[ndx] = ureg.convert(values[ndx].magnitude, values[ndx].units, args_as_uc[ndx])
+            elif isinstance(values[ndx], collections.abc.MutableSequence):
+                for i, val in enumerate(values[ndx]):
+                    if isinstance(val, ureg.Quantity):
+                        values[ndx][i] = ureg.convert(val.magnitude, val.units, args_as_uc[ndx])
```

Comment:
Another tiny performance improvement
```suggestion
            value = values[ndx]
            if isinstance(value, ureg.Quantity):
                values[ndx] = ureg.convert(value.magnitude, value.units, args_as_uc[ndx])
            elif isinstance(value, MutableSequence):
                for i, val in enumerate(value):
                     if isinstance(val, ureg.Quantity):
                        value[i] = ureg.convert(val.magnitude, val.units, args_as_uc[ndx])
```